gazelle/parser.go

```diff
@@ -0,0 +1,88 @@
+package gazelle
+
+import (
+	"log"
+
+	"path/filepath"
+
+	"github.com/evanw/esbuild/pkg/api"
+)
+
+type Parser struct {
+}
+
+func NewParser() *Parser {
+	p := &Parser{}
+	return p
+}
+
+// filenameToLoader takes in a filename, e.g. myFile.ts,
+// and returns the appropriate esbuild loader for that file.
+func filenameToLoader(filename string) api.Loader {
+	ext := filepath.Ext(filename)
+	switch ext {
+	case ".ts":
+		return api.LoaderTS
+	case ".tsx":
+		return api.LoaderTSX
+	case ".js":
+		return api.LoaderJSX
+	case ".jsx":
+		return api.LoaderJSX
+	default:
+		return api.LoaderTS
+	}
+}
+
+// ParseImports returns all the imports from a file
+// after parsing it.
+func (p *Parser) ParseImports(filePath, source string) []string {
+	imports := []string{}
+
+	// Construct an esbuild plugin that pulls out all the imports.
+	plugin := api.Plugin{
+		Name: "GetImports",
```

Comment:
`GetImports` is not a const somewhere upstream?

Reply:
I think it's just for debugging essentially. I assume esbuild will put this in logs if there is an error or plugins are listed somewhere etc. I think @alexeagle originally wrote this part.